plugins/secrets/index.js

```diff
@@ -42,6 +49,11 @@ exports.register = (server, options, next) => {
                     }
 
                     return user.getPermissions(pipeline.scmUrl).then(permissions => {
+                        if (permission === 'admin' && !permissions.admin) {
```

Comment:
You can do this so you don't need to split into multiple checks for admin and push:
```js 
if (!permisions[requiredPerm]) {
   throw boom.forbidden(`User ${username} does not have ${requiredPerm} access to this repo`)
``` 
where `requiredPerm` is the param passed in.